src/conways.rs

```diff
@@ -0,0 +1,126 @@
+const COLUMNS: usize = 32;
+const ROWS: usize = 32;
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum CellState {
+    Alive,
+    Dead,
+}
+
+#[derive(Clone, Copy)]
+pub struct Position(pub usize, pub usize);
+
+pub struct Conways {
+    pub grid: Vec<Vec<CellState>>,
+}
+
+impl Conways {
+
+    // Using this function in the test
+    #[allow(dead_code)]
+    pub fn default() -> Self {
+        Self {
+            grid: vec![vec![CellState::Dead; COLUMNS]; ROWS]
+        }
+    }
+
+    pub fn from(alive_cells: Vec<Position>) -> Self {
+        let mut grid = vec![vec![CellState::Dead; COLUMNS]; ROWS];
+        
+        for Position(x, y) in alive_cells {
+            if x < ROWS && y < COLUMNS {
+                grid[x][y] = CellState::Alive;
+            }
+        }
+        
+        Self { grid }
+    }
+
+    pub fn update_cells(&mut self) {
+        let mut new_grid = self.grid.clone();
+    
+        for (x, _) in self.grid.iter().enumerate() {
+            for (y, _) in self.grid.iter().enumerate() {
+                let neighbors = self.count_neighbors(Position(x, y));
```

Comment:
This wouldn't work if the ROWS != COLUMNS

Reply:
Yes, you are right! I came up with this idea, but I'm not sure if it's the correct one:
`for (y, _) in self.grid.first().unwrap_or(&vec![]).iter().enumerate()`
For my understanding, the `unwrap_or` will never panic since we are passing a value when `first()` returns `None`.

Reply:
Try using something like this if you want to avoid using `unwrap`
```Rust
            let Some(cols) = self.grid.first() else {
                return;
            };
            for (y, _) in cols.iter().enumerate() {
                //code goes here
            }
```